### PR TITLE
chore(pebbledb): remove isForceSync flag

### PR DESCRIPTION
**Pull request by melekes**
Downstream dependencies should use `SetSync`/`DeleteSync` to force
flushing OS buffers to disk.

### CODE REVIEW

**Review by jmalicevic — APPROVED** [review]
In general mixing concerns within one PR might be less desired but these are all minor changes so OK to merge it as is.

File: README.md [hunk 1]
```diff
@@ -58,6 +58,12 @@ Go 1.21+
   performance, and includes advanced features such as serializable ACID
   transactions, write batches, compression, and more.
 
+- **[PebbleDB](https://github.com/cockroachdb/pebble) [experimental]:** Pebble
```
Comment (jmalicevic) [inline README.md:61]: Maybe add here the note existing in ForceSync that people using the SDK might have a problem with this.

File: README.md [hunk 2]
```diff
@@ -58,6 +58,15 @@ Go 1.21+
...
+  is a LevelDB/RocksDB inspired key-value store focused on performance and
+  internal usage by CockroachDB. Pebble inherits the RocksDB file formats and a
+  few extensions such as range deletion tombstones, table-level bloom filters,
+  and updates to the MANIFEST format.
+
+  XXX: there are reports of broken upgrade process when using [Cosmos
```
Comment (mzabaluev) [inline README.md:67]:
```suggestion
  CAVEAT: there are reports of broken upgrade process when using [Cosmos
```

XXX is a jargon more appropriate for the code docs, I think.

File: README.md (hunk 1 above)
Comment (melekes) [reply README.md:61]: created https://github.com/cometbft/cometbft-db/issues/126

**Pull request merged**